### PR TITLE
Changes to UI after app installation is complete

### PR DESCRIPTION
**Pull request by jyvenugo**
-Installed app launches if the 'Launch when ready' checkbox was checked and Msix app installer closes.
-If the checkbox is not checked, the launch window appears displaying the 'Launch' button.
-Clicking on Launch launches the app.
-Created InstallComplete handlers.
-Updated path so icon appears in control panel.
-Installed apps display only the publisher name now in 'Add or remove programs'.

### CODE REVIEW

File: preview/Win7Msix/Win7MSIXInstaller/InstallUI.hpp [hunk 1]
```diff
@@ -118,3 +141,8 @@ BOOL ChangeText(HWND parentHWnd, std::wstring displayText, std::wstring  message
 //
 // PURPOSE: Increment the progress bar one tick based on preset tick
 void UpdateProgressBar();
+
+// FUNCTION: SendInstallCompleteMsg
+//
+// PURPOSE: Sends the WM_INSTALLCOMPLETE_MSG message to the main window when app installation is complete
+void SendInstallCompleteMsg();
```
Comment (wcheng-msft) [inline preview/Win7Msix/Win7MSIXInstaller/InstallUI.hpp]: >SendInstallCompleteMsg [](start = 5, length = 22)

can these be made members of the UI class? I hadn't gotten around to moving all the global stuff into the class, which is why there's so much stuff left as global. I'd rather not add more of this.

File: preview/Win7Msix/Win7MSIXInstaller/InstallUI.cpp [hunk 2]
```diff
@@ -439,6 +484,24 @@ BOOL CreateCancelButton(HWND parentHWnd, RECT parentRect) {
 	return TRUE;
 }
 
+BOOL CreateLaunchButton(HWND parentHWnd, RECT parentRect) {
```
Comment (wcheng-msft) [inline preview/Win7Msix/Win7MSIXInstaller/InstallUI.cpp]: > { [](start = 57, length = 2)

style-wise let's be consistent on making this on a separate line by itself

File: preview/Win7Msix/Win7MSIXInstaller/InstallUI.hpp [hunk 3]
```diff
@@ -25,22 +29,33 @@ class UI
...
+		m_launchAppEvent = CreateEvent(NULL, FALSE, FALSE, NULL);
+	}
     
 public:
     HRESULT DisplayPackageInfo(HWND hWnd, RECT windowRect, std::wstring& displayText, std::wstring& messageText);
     int CreateInitWindow(HINSTANCE hInstance, int nCmdShow, const std::wstring& windowClass, const std::wstring& title);
 
     void SetButtonClicked() { SetEvent(m_buttonClickedEvent); }
+    void SetLaunchButtonClicked() { SetEvent(m_launchAppEvent); }
+    HANDLE getLaunchButtonEvent()
```
Comment (wcheng-msft) [inline preview/Win7Msix/Win7MSIXInstaller/InstallUI.hpp]: >g [](start = 11, length = 1)

caps for method names?

File: preview/Win7Msix/Win7MSIXInstaller/InstallUI.hpp [hunk 4]
```diff
@@ -25,22 +29,33 @@ class UI
...
+		m_buttonClickedEvent = CreateEvent(NULL, FALSE, FALSE, NULL);
+		m_launchAppEvent = CreateEvent(NULL, FALSE, FALSE, NULL);
+	}
     
 public:
     HRESULT DisplayPackageInfo(HWND hWnd, RECT windowRect, std::wstring& displayText, std::wstring& messageText);
     int CreateInitWindow(HINSTANCE hInstance, int nCmdShow, const std::wstring& windowClass, const std::wstring& title);
 
     void SetButtonClicked() { SetEvent(m_buttonClickedEvent); }
+    void SetLaunchButtonClicked() { SetEvent(m_launchAppEvent); }
```
Comment (wcheng-msft) [inline preview/Win7Msix/Win7MSIXInstaller/InstallUI.hpp]: >SetLaunchButtonClicked [](start = 9, length = 22)

i don't see this being called. did i miss it or is there a future task to call this?

Comment (jyvenugo) [reply preview/Win7Msix/Win7MSIXInstaller/InstallUI.hpp]: I created the event so the UI will wait without exiting while waiting for the launch button to be clicked at the same time using MsgWaitForMultipleObjects. The event is not being set anywhere.

File: preview/Win7Msix/Win7MSIXInstaller/InstallUI.hpp (hunk 1 above)
Comment (jyvenugo) [reply preview/Win7Msix/Win7MSIXInstaller/InstallUI.hpp]: Moved UI controls and methods to members of UI class. Updated call locations.

File: preview/Win7Msix/Win7MSIXInstaller/InstallUI.cpp (hunk 2 above)
Comment (jyvenugo) [reply preview/Win7Msix/Win7MSIXInstaller/InstallUI.cpp]: Moved bracket to new line.

File: preview/Win7Msix/Win7MSIXInstaller/InstallUI.hpp (hunk 3 above)
Comment (jyvenugo) [reply preview/Win7Msix/Win7MSIXInstaller/InstallUI.hpp]: Updated method name.

File: preview/Win7Msix/Win7MSIXInstaller/InstallUI.hpp (hunk 4 above)
Comment (wcheng-msft) [reply preview/Win7Msix/Win7MSIXInstaller/InstallUI.hpp]: how does it work currently if you aren't setting the event anywhere? i guess what i was asking was did you mean to call it somewhere and forget to add it? or are you planning on calling it somewhere in the future checkin as part of a different task? or are you finished with this and intend for it never to be set?

---
In reply to: [264356252](https://github.com/Microsoft/msix-packaging/pull/59#discussion_r264356252) [](ancestors = 264356252)

File: preview/Win7Msix/Win7MSIXInstaller/InstallUI.cpp [hunk 5]
```diff
@@ -145,8 +145,37 @@ LRESULT CALLBACK WndProc(HWND hWnd, UINT message, WPARAM wParam, LPARAM lParam)
...
+        if (g_launchCheckBoxState) {
+            ui->LaunchInstalledApp(); // launch app
+            DestroyWindow(hWnd); // close msix app installer
+        }
+        else
+        {
+            //wait for user to click launch button or close the window
+            while (true)
+            {
+                MsgWaitForMultipleObjects(0, NULL, FALSE, INFINITE, QS_ALLINPUT);
```
Comment (wcheng-msft) [inline preview/Win7Msix/Win7MSIXInstaller/InstallUI.cpp]: >MsgWaitForMultipleObjects [](start = 16, length = 25)

even though you're waiting infinitely for 0 handles, you should still check the return value. It should only ever return one value, but you should still check it. Also the open brace (for the while) should be on a line by itself

**Pull request merged**